python/lib/monitor/dmod/monitor/que_monitor.py

```diff
@@ -38,989 +20,420 @@
+                             redis_port=redis_port, redis_pass=redis_pass, **kwargs)
+        if 'type' in kwargs:
+            key_prefix = RedisBackedJobManager.get_key_prefix(environment_type=kwargs['type'])
+        else:
+            key_prefix = RedisBackedJobManager.get_key_prefix()
+        self._active_jobs_set_key = self.keynamehelper.create_key_name(key_prefix, 'active_jobs')
 
-    # let the jobs run to finish
-    time.sleep(30)
+    def get_jobs_to_monitor(self) -> List[Job]:
+        """
+        Get a list of the jobs currently needing monitoring.
 
-    # runningQ_restart = False
-    runningQ_restart = True
+        This will likely be something akin to "active" jobs, but the particular conditions should be documented for each
+        implementation.
 
-    if runningQ_restart == True:
-        q.restart_running_jobQ()
+        Returns
+        -------
+        List[Job]
+            A list of the job objects corresponding to executing jobs within the runtime that need to be monitored.
+        """
+        active_jobs = []
+        for active_job_redis_key in self.redis.smembers(self._active_jobs_set_key):
+            # If key exists
+            if self.redis.exists(active_job_redis_key) == 1:
+                serialized_job = json.loads(self.redis.get(active_job_redis_key))
+                job = Job.factory_init_from_deserialized_json(serialized_job)
+                if isinstance(job, Job):
+                    active_jobs.append(job)
+        return active_jobs
 
-    else:
-        # TODO double check the interaction between this function call and q.check_job_state()
-        state_dict = q.build_state_list_reqid_set(service_state_list)
-        q.service_actions(runningJobList, state_dict)
 
-    nodeList = q.get_node_info()
-    q.check_available_resources()
+class RedisDockerSwarmMonitor(DockerSwarmMonitor, RedisBackedMonitor):
+    """
+    Concrete subtype of both ::class:`DockerSwarmMonitor` and ::class:`RedisBackedMonitor`.
 
-    while True:
-        time.sleep(30)
-        q.check_system_state()
+    Concrete ::class:`Monitor` implementation inheriting from ::class:`DockerSwarmMonitor` (for runtime interaction and
+    monitoring logic) and ::class:`RedisBackedMonitor` (for job monitoring eligibility/filtering logic).
+    """
 
-    print("end of que_monitor")
+    def __init__(self, resource_pool: str,
+                 docker_client: docker.from_env() = None, api_client: docker.APIClient() = None,
+                 redis_host: Optional[str] = None, redis_port: Optional[int] = None,
+                 redis_pass: Optional[str] = None, **kwargs):
+        DockerSwarmMonitor.__init__(self, docker_client, api_client)
+        RedisBackedMonitor.__init__(self, resource_pool=resource_pool, redis_host=redis_host, redis_port=redis_port,
+                                    redis_pass=redis_pass, **kwargs)
 
-if __name__ == "__main__":
```

Comment:
Brutal diff 🤯

Reply:
I don't think we want that here, do we?  We really don't want something like this anywhere except a package _\_\_main\_\_.py_ file, right?

Reply:
Sorry my comment wasn't clear.  I have no issues with these changes -- the comment was on how the Diff wasn't really useful with how significant things got refactored haha.